tests/runtest.py

```diff
@@ -705,6 +715,103 @@ def correct_line_endings(host_os, test_location, root=True):
             with open(test_location, 'w') as file_handle:
                 file_handle.write(content)
 
+def setup_coredump_generation():
+    if host_os != "Linux" and host_os != "OSX":
+        print("CoreDump generation not enabled due to unsupported OS")
+        return
+
+    if host_os == "OSX" and subprocess.check_output("sysctl -n kern.corefile", shell=True).rstrip() != "core.%P":
```

Comment:
Maybe we should also be checking `/proc/sys/kernel/core_pattern`?

We could also just set `core_pattern` for the current run, to some random tmp directory...